src/rgw/rgw_lua_background.cc

```diff
@@ -153,7 +154,44 @@ void Background::run() {
       }
       ldpp_dout(dpp, 10) << "Lua background thread resumed" << dendl;
     }
-    
+    std::size_t max_memory = cct->_conf->rgw_lua_max_memory_per_state;
+    std::uint64_t max_runtime = cct->_conf->rgw_lua_max_runtime_per_state;
+    if (lguard->get_max_memory() != max_memory) {
```

Comment:
nit: dont think this is needed, the `set_max_memory()` function handle this case